src/OrchardCore/Orchard.Environment.Shell.Abstractions/ShellSettings.cs

```diff
@@ -11,7 +11,6 @@ namespace Orchard.Environment.Shell
     /// </summary>
     public class ShellSettings
     {
-        private TenantState _tenantState = TenantState.Invalid;
```

Comment:
Maybe it should be kept and parsed in the ctor instead of the getter. This way the setter and the getter manipulate the Enum.